include/engine/datafacade/contiguous_internalmem_datafacade.hpp

```diff
@@ -1008,6 +1058,36 @@ class ContiguousInternalMemoryDataFacade : public BaseDataFacade
                     m_lane_description_offsets[lane_description_id + 1]);
     }
 };
+
+template <typename AlgorithmT> class ContiguousInternalMemoryDataFacade;
+
+template <>
+class ContiguousInternalMemoryDataFacade<algorithm::CH>
+    : public ContiguousInternalMemoryDataFacadeBase,
+      public ContiguousInternalMemoryAlgorithmDataFacade<algorithm::CH>
```

Comment:
Yes this makes it possible to use both the shared functions and the algorithm specific functions on the same object instead of needing to keep track of which is which.